core/bin/zksync_api/src/api_server/rest/v01/api_impl.rs

```diff
@@ -65,27 +66,61 @@ impl ApiV01 {
                 .expect("TickerConfig::liquidity_volume must be positive"),
         );
 
+        let ticker_config = zksync_config::configs::TickerConfig::from_env();
+        let unconditionally_valid_addresses = ticker_config
+            .unconditionally_valid_tokens
+            .iter()
+            .map(|h160| h160.to_string())
+            .collect::<Vec<String>>();
+
         let mut storage = self_.access_storage().await?;
-        let mut tokens = storage
+        let tokens = storage
             .tokens_schema()
-            .load_tokens_by_market_volume(liquidity_volume)
+            .load_tokens()
             .await
             .map_err(Self::db_error)?;
 
-        // Add ETH for tokens allowed for fee
-        // Different APIs have different views on how to represent ETH in their system.
-        // But ETH is always allowed to pay fee, and in all cases it should be on the list.
-
-        if tokens.get(&TokenId(0)).is_none() {
-            let eth = Token::new(TokenId(0), Default::default(), "ETH", 18, TokenKind::ERC20);
-            tokens.insert(eth.id, eth);
-        }
+        // I know this seems like a very inefficient way of doing it, when we could just make a left join in sql and in fact I've got that code stashed for two reasons:
+        // 1. the storage manager should not really have any knowledge of the unconditionally valid tokens
+        // 2. and more importantly, I couldn't make it work, due to limitations of myself and/or sqlx (the 'token.address IN ()' seems to be problematic when not dealing with hardcoded values)
```

Comment:
what kind of errors did you have? I also experienced some issues trying to create my own query with sqlx, but now I think that its working.

Reply:
hm true that